paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -122,6 +122,12 @@ public class PaparazziPlugin @Inject constructor(
       it.delete(files)
     }
 
+    // We need to pull target sdk as defined from DSL otherwise it gets set to some default value when resolving during [onVariants]
+    var targetSdkVersion: String? = null
+    extension.finalizeDsl { androidExtension ->
+      targetSdkVersion = androidExtension?.targetSdkVersion() ?: DEFAULT_COMPILE_SDK_VERSION.toString()
+    }
+
```

Comment:
This is the bulk of the change where `BaseExtension` is replaced with this dsl extension handling.